buildSrc/src/main/kotlin/me/vkryl/task/TaskFunctions.kt

```diff
@@ -54,13 +54,10 @@ fun writeToFile(path: String, mkdirs: Boolean = true, block: (Writer) -> Unit) {
   }
 
   if (file.exists()) {
-    if (!areFileContentsIdentical(file, outFile)) {
-      copyOrReplace(outFile, file)
-    }
-    outFile.delete()
-  } else {
-    outFile.renameTo(file)
```

Comment:
Do we really need a content identity check here? Because the error `The requested operation cannot be performed on a file with a user-mapped section open` occurs here when the `areFileContentsIdentical` and `copyOrReplace` functions are called one after another.

Reply:
> Do we really need a content identity check here? Because the error `The requested operation cannot be performed on a file with a user-mapped section open` occurs here when the `areFileContentsIdentical` and `copyOrReplace` functions are called one after another.

Yes. IDE behaves much better when the old file isn't replaced, as it doesn't do full reindexing of it and also file creation & modification time keep unchanged.

Could you try refactoring `areFileContentsIdentical` to this:

```
fun areFileContentsIdentical(a: File, b: File): Boolean {
  val areIdentical: Boolean
  FileChannel.open(a.toPath(), StandardOpenOption.READ).use { fileChannelA ->
    FileChannel.open(b.toPath(), StandardOpenOption.READ).use { fileChannelB ->
      val mapA = fileChannelA.map(FileChannel.MapMode.READ_ONLY, 0, fileChannelA.size())
      val mapB = fileChannelB.map(FileChannel.MapMode.READ_ONLY, 0, fileChannelB.size())
      areIdentical = mapA == mapB
    }
  }
  return areIdentical
}
```

If that doesn't help, just a slight delay before `copyOrReplace(outFile, file)`, e.g. via `Thread.sleep`?

Reply:
It seems that `Thread.sleep(300)` works.  For `File.delete()` it must be repeated several times and call `System.gc()`.